pkg/webhook/nodenetworkconfigurationpolicy/conditions.go

```diff
@@ -0,0 +1,107 @@
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+const (
+	TimestampLabelKey = "nmstate.io/webhook-mutating-timestamp"
+)
+
+var log = logf.Log.WithName("webhook/nodenetworkconfigurationpolicy/conditions")
+
+type mutator func(nmstatev1alpha1.NodeNetworkConfigurationPolicy) nmstatev1alpha1.NodeNetworkConfigurationPolicy
+
+func deleteConditions(policy nmstatev1alpha1.NodeNetworkConfigurationPolicy) nmstatev1alpha1.NodeNetworkConfigurationPolicy {
+	policy.Status.Conditions = nmstatev1alpha1.ConditionList{}
+	return policy
+}
+
+func setConditionsUnknown(policy nmstatev1alpha1.NodeNetworkConfigurationPolicy) nmstatev1alpha1.NodeNetworkConfigurationPolicy {
+	for _, conditionType := range nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionTypes {
+		policy.Status.Conditions.Set(
+			conditionType,
+			corev1.ConditionUnknown,
+			"", "")
+	}
+	return policy
+}
+
+func resetTimestampAnnotation(policy nmstatev1alpha1.NodeNetworkConfigurationPolicy) nmstatev1alpha1.NodeNetworkConfigurationPolicy {
+	value := strconv.FormatInt(time.Now().UnixNano(), 10)
+	if policy.ObjectMeta.Annotations == nil {
+		policy.ObjectMeta.Annotations = map[string]string{}
+	}
+	policy.ObjectMeta.Annotations[TimestampLabelKey] = value
+	return policy
+}
+
+func mutatePolicy(policy nmstatev1alpha1.NodeNetworkConfigurationPolicy, mutators ...mutator) nmstatev1alpha1.NodeNetworkConfigurationPolicy {
+	for _, mutator := range mutators {
+		policy = mutator(policy)
+	}
+	return policy
+}
+func mutatePolicyHandler(neededMutationFor func(nmstatev1alpha1.NodeNetworkConfigurationPolicy) bool, mutators ...mutator) admission.HandlerFunc {
+	return func(ctx context.Context, req webhook.AdmissionRequest) webhook.AdmissionResponse {
+		original := req.Object.Raw
+		policy := nmstatev1alpha1.NodeNetworkConfigurationPolicy{}
+		err := json.Unmarshal(original, &policy)
+		if err != nil {
+			return admission.Errored(http.StatusInternalServerError, errors.Wrapf(err, "failed decoding policy: %s", string(original)))
+		}
+		if !neededMutationFor(policy) {
+			return admission.Allowed("mutation not needed")
+		}
+		policy = mutatePolicy(policy, mutators...)
+		current, err := json.Marshal(policy)
+		if err != nil {
+			return admission.Errored(http.StatusInternalServerError, errors.Wrapf(err, "failed encoding policy: %+v", policy))
+		}
+		response := admission.PatchResponseFromRaw(original, current)
+		log.Info(fmt.Sprintf("webhook response: %+v", response))
+		return response
+	}
+}
+
+func deleteConditionsHook() *webhook.Admission {
+	return &webhook.Admission{
+		Handler: admission.HandlerFunc(
+			mutatePolicyHandler(
+				func(nmstatev1alpha1.NodeNetworkConfigurationPolicy) bool {
+					// Always delete conditions when nncp is changed
+					return true
+				},
+				deleteConditions,
+				resetTimestampAnnotation)),
```

Comment:
I would use a separate hook for the timestamp anotation. Then you can make this helper Handler function simpler.

Reply:
maybe is even better to use different annotation one for nncp and other nncp/status